pkg/location/circlecicontext.go

```diff
@@ -0,0 +1,72 @@
+package location
+
+import (
+	"github.com/ovotech/cloud-key-rotator/pkg/cred"
+
+	"github.com/CircleCI-Public/circleci-cli/api"
+	graphqlclient "github.com/CircleCI-Public/circleci-cli/client"
+)
+
+// CircleCIContext type
+type CircleCIContext struct {
+	ContextID   string
+	KeyIDEnvVar string
+	KeyEnvVar   string
+}
+
+func (circleContext CircleCIContext) Write(serviceAccountName string, keyWrapper KeyWrapper, creds cred.Credentials) (updated UpdatedLocation, err error) {
+	logger.Info("Starting CircleCI context env var updates")
+
+	gqlclient := graphqlclient.NewClient(
+		"https://circleci.com",
+		"graphql",
+		creds.CircleCIAPIToken,
+		false,
+	)
+
+	provider := keyWrapper.KeyProvider
+	contextID := circleContext.ContextID
+
+	var keyEnvVar string
+	var idValue bool
+	if keyEnvVar, err = getVarNameFromProvider(provider, circleContext.KeyEnvVar, idValue); err != nil {
+		return
```

Comment:
Do we have an explicit convention to use named return variables and implicit returns? My personal opinion is that this makes values harder to track throughout a function, rather than e.g.

```
func (circleContext CircleCIContext) Write(serviceAccountName string, keyWrapper KeyWrapper, creds cred.Credentials) (*UpdatedLocation, error) {
*....)
	if keyEnvVar, err := getVarNameFromProvider(provider, circleContext.KeyEnvVar, idValue); err != nil {
		return nil, err

(...)
```

What do you think?

Reply:
We don't have anything in place in terms of conventions around named variables and returns.

I've definitely heard that point of view on several occasions before. I can see the benefit of explicit returns in larger functions for tracking different paths, though I try to keep my functions small enough that tracking should be straightforward. Perhaps some of my functions could actually do with refactoring to trim them down. In small functions I find it easy enough to check what's returned from the function signature. It also helps when the majority of returns are just for `nil, err`, though, I admit, implicit returns don't exactly help readability for other people..

I like the fact that named return variables are zero-valued upon entry into the function, it often saves some code/time.

I take your point though. Thanks for spending some time on this 👍